### PR TITLE
Allow batch publishing for emit all process

### PR DESCRIPTION
**Pull request by joskfg**

### CODE REVIEW

File: src/Console/Commands/EmitAllEvents.php [hunk 1]
```diff
@@ -33,16 +34,18 @@ public function handle()
...
 
         DomainEvent::on($databaseConnection)->cursor()
+            ->chunk($batchSize)
             ->each(
-                function ($domainEvent) use ($queueConnection, $bar) {
-                    SendDomainEvents::dispatch($domainEvent->message)
+                function ($domainEvents) use ($queueConnection, $bar) {
+                    SendDomainEvents::dispatch(SendDomainEvents::NO_RETRIES, ...$domainEvents->pluck('message'))
                         ->onConnection($queueConnection);
                     $bar->advance();
```
Comment (joskfg) [inline src/Console/Commands/EmitAllEvents.php]: This should be fixed indicating the batchSize.

File: src/Console/Commands/EmitAllEvents.php [hunk 2]
```diff
@@ -15,7 +15,8 @@ class EmitAllEvents extends Command
      */
     protected $signature = 'event-sourcing:emit-all
         {queueConnection=database : Queue connection to be used to send the messages}
-        {--unbufferedConnection= : Indicate the unbuffered connection (MySQL) for large amount of events}';
+        {--unbufferedConnection= : Indicate the unbuffered connection (MySQL) for large amount of events}
+        {--batchSize=1 : Indicate the amount of events to be sent per publish. Increase for higher throughput}';
```
Comment (jeurrutia) [inline src/Console/Commands/EmitAllEvents.php:19]: We can put a batch size higher by default, WDYT?

Comment (joskfg) [reply src/Console/Commands/EmitAllEvents.php:19]: There is no good default value, it depends on your use case, so I prefer to use the most standard value here.

File: src/Jobs/SendDomainEvents.php [hunk 3]
```diff
@@ -67,8 +68,8 @@ public function handle(AmqpMiddleware $amqpMiddleware, Dispatcher $dispatcher, L
 
     protected function sendEvent(): void
     {
-        if (!$this->amqpMiddleware->store($this->eventMessage)) {
-            $errorMessage = "The event could't be sent. Retrying message: " . json_encode($this->eventMessage);
+        if (!$this->amqpMiddleware->store(...$this->eventMessages)) {
+            $errorMessage = "The event could't be sent. Retrying message: " . json_encode($this->eventMessages);
```
Comment (jeurrutia) [inline src/Jobs/SendDomainEvents.php:72]: There is a typo (couldn't)

**Review by jeurrutia — APPROVED** [review]

**Review by xaviapa — APPROVED** [review]

**Pull request merged**